File: ios/Classes/TangentSdkPlugin.swift
import Flutter
import UIKit

public class TangentSdkPlugin: NSObject, FlutterPlugin {

    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(
            name: "com.tangent_sdk/billing_issue",
            binaryMessenger: registrar.messenger()
        )
        let instance = TangentSdkPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "checkBillingIssue":
            if #available(iOS 15.0, *) {
                let handler = BillingIssueHandler()
                handler.checkBillingIssue(result: result)
            } else {
                result([
                    "state": "normal",
                    "managementURL": "https://apps.apple.com/account/subscriptions",
                ])
            }
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}

File: android/src/main/kotlin/com/tangent_sdk/TangentSdkPlugin.kt
package com.tangent_sdk

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class TangentSdkPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel
    private val billingIssueHandler = BillingIssueHandler()

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "com.tangent_sdk/billing_issue")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "checkBillingIssue" -> billingIssueHandler.checkBillingIssue(result)
            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
